extensions/parquet/table/src/main/java/io/deephaven/parquet/table/transfer/DictEncodedStringTransferBase.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.parquet.table.transfer;
+
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.table.ColumnSource;
+import org.jetbrains.annotations.NotNull;
+
+import java.nio.IntBuffer;
+import java.util.function.Supplier;
+
+/**
+ * Base class for reading dictionary-encoded string arrays and vectors. This class updates the {@link StringDictionary}
+ * with all the strings it encounters and generates an IntBuffer of dictionary position values. This class extends
+ * {@link PrimitiveArrayAndVectorTransfer} to manage the dictionary positions in an {@link IntBuffer} similar to an Int
+ * array/vector column.
+ */
+abstract public class DictEncodedStringTransferBase<T>
+        extends PrimitiveArrayAndVectorTransfer<T, IntBuffer, IntBuffer> {
+    private boolean pageHasNull;
+    private final StringDictionary dictionary;
+    private final int nullValue; // The value to store in buffer for null strings
+    private IntBuffer dictEncodedValues;
```

Comment:
Seems to me like we might be better off with an array of ints, since we're bulk-putting this into another buffer.